freqtrade/leverage/liquidation_price.py

```diff
@@ -0,0 +1,357 @@
+                interest=interest,
+                taker_fee_rate=taker_fee_rate,
+                position_assets=position_assets,
+            )
+    elif exchange_name.lower() == "ftx":
+        return ftx(open_rate, is_short, leverage, trading_mode, collateral)
+    elif exchange_name.lower() == "kraken":
+        return kraken(open_rate, is_short, leverage, trading_mode, collateral)
+    raise OperationalException(f"liquidation_price is not implemented for {exchange_name}")
+
+
+def exception(
+    exchange: str,
+    trading_mode: TradingMode,
+    collateral: Collateral,
+):
+    """
+    Raises an exception if exchange used doesn't support desired leverage mode
+    :param exchange: Name of the exchange
+    :param trading_mode: spot, margin, futures
+    :param collateral: cross, isolated
+    """
+
+    raise OperationalException(
+        f"{exchange} does not support {collateral.value} Mode {trading_mode.value} trading ")
+
+
+def binance(
+    open_rate: float,
+    is_short: bool,
+    leverage: float,
+    trading_mode: TradingMode,
+    mm_ratio: float,
+    collateral: Collateral,
+    maintenance_amt: float,
+    wallet_balance: float,
+    position: float,
+    mm_ex_1: float,
+    upnl_ex_1: float,
+):
+    """
+    MARGIN: https://www.binance.com/en/support/faq/f6b010588e55413aa58b7d63ee0125ed
+    PERPETUAL: https://www.binance.com/en/support/faq/b3c689c1f50a44cabb3a84e663b81d93
+
+    :param open_rate: (EP1) Entry Price of position (one-way mode)
+    :param is_short: true or false
+    :param leverage: leverage in float
+    :param trading_mode: SPOT, MARGIN, FUTURES
+    :param mm_ratio: (MMR) Maintenance margin rate of position (one-way mode)
+    :param collateral: CROSS, ISOLATED
+    :param maintenance_amt: (CUM) Maintenance Amount of position (one-way mode)
+    :param position: Absolute value of position size (one-way mode)
+    :param wallet_balance: (WB)
+        Cross-Margin Mode: crossWalletBalance
+        Isolated-Margin Mode: isolatedWalletBalance
+            TMM=0, UPNL=0, substitute the position quantity, MMR, cum into the formula to calculate.
+            Under the cross margin mode, the same ticker/symbol,
+            both long and short position share the same liquidation price except in isolated mode.
+            Under the isolated mode, each isolated position will have different liquidation prices
+            depending on the margin allocated to the positions.
+    :param mm_ex_1: (TMM)
+        Cross-Margin Mode: Maintenance Margin of all other contracts, excluding Contract 1
+        Isolated-Margin Mode: 0
+    :param upnl_ex_1: (UPNL)
+        Cross-Margin Mode: Unrealized PNL of all other contracts, excluding Contract 1.
+        Isolated-Margin Mode: 0
+    """
+    side_1 = -1 if is_short else 1
+    position = abs(position)
+    cross_vars = upnl_ex_1 - mm_ex_1 if collateral == Collateral.CROSS else 0.0
+
+    if trading_mode == TradingMode.MARGIN and collateral == Collateral.CROSS:
```

Comment:
Is this correct, or did you mean `or` here? (i don't think we support Margin or Cross for binance).